thinc/util.py

```diff
@@ -257,14 +257,20 @@ def smooth_one_hot(X: Floats2d, label_smoothing: float) -> Floats2d:
     """
     Apply label-smoothing to one-hot array.
     """
-    if not 0.0 <= label_smoothing < 0.5:
+    if label_smoothing < 0.0:
+        raise ValueError(
+            "Label-smoothing parameter has to be greater than or equal to 0"
+        )
+    n_classes = X.shape[1]
+    max_smooth = (n_classes - 1) / n_classes
+    if label_smoothing >= max_smooth:
```

Comment:
I think like `to_categorical`, we also need a check here to ensure that the number of classes is larger than one. Otherwise, `max_smooth == 0.0` and even with label_smoothing==0.0, we'll get the strange error message _For 1 classes label_smoothing parameter has to be less than 0, but found 0._